core/embed/rust/src/ui/model_mercury/layout.rs

```diff
@@ -550,6 +490,216 @@ extern "C" fn new_confirm_homescreen(n_args: usize, args: *const Obj, kwargs: *m
 }
 
+extern "C" fn new_confirm_reset(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let recovery: bool = kwargs.get(Qstr::MP_QSTR_recovery)?.try_into()?;
+
+        let flow = flow::confirm_reset::new_confirm_reset(recovery)?;
+        Ok(LayoutObj::new_root(flow)?.into())
+    };
+    unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
+}
+
+extern "C" fn new_confirm_set_new_pin(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: TString = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let description: TString = kwargs.get(Qstr::MP_QSTR_description)?.try_into()?;
+
+        let flow = flow::confirm_set_new_pin::new_set_new_pin(title, description)?;
+        Ok(LayoutObj::new_root(flow)?.into())
+    };
+    unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
+}
+
+extern "C" fn new_confirm_output(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: Option<TString> = kwargs.get(Qstr::MP_QSTR_title)?.try_into_option()?;
+        let subtitle: Option<TString> = kwargs.get(Qstr::MP_QSTR_subtitle)?.try_into_option()?;
+
+        let account: Option<TString> = kwargs.get(Qstr::MP_QSTR_account)?.try_into_option()?;
+        let account_path: Option<TString> =
+            kwargs.get(Qstr::MP_QSTR_account_path)?.try_into_option()?;
+
+        let br_name: TString = kwargs.get(Qstr::MP_QSTR_br_name)?.try_into()?;
+        let br_code: u16 = kwargs.get(Qstr::MP_QSTR_br_code)?.try_into()?;
+
+        let message: Obj = kwargs.get(Qstr::MP_QSTR_message)?;
+        let amount: Option<Obj> = kwargs.get(Qstr::MP_QSTR_amount)?.try_into_option()?;
+
+        let chunkify: bool = kwargs.get_or(Qstr::MP_QSTR_chunkify, false)?;
+        let text_mono: bool = kwargs.get_or(Qstr::MP_QSTR_text_mono, true)?;
+
+        let address: Option<Obj> = kwargs.get(Qstr::MP_QSTR_address)?.try_into_option()?;
+        let address_title: Option<TString> =
+            kwargs.get(Qstr::MP_QSTR_address_title)?.try_into_option()?;
+
+        let summary_items: Option<Obj> =
+            kwargs.get(Qstr::MP_QSTR_summary_items)?.try_into_option()?;
+        let fee_items: Option<Obj> = kwargs.get(Qstr::MP_QSTR_fee_items)?.try_into_option()?;
+
+        let summary_title: Option<TString> =
+            kwargs.get(Qstr::MP_QSTR_summary_title)?.try_into_option()?;
+        let summary_br_name: Option<TString> = kwargs
+            .get(Qstr::MP_QSTR_summary_br_name)?
+            .try_into_option()?;
+        let summary_br_code: Option<u16> = kwargs
+            .get(Qstr::MP_QSTR_summary_br_code)?
+            .try_into_option()?;
+
+        let cancel_text: Option<TString> =
+            kwargs.get(Qstr::MP_QSTR_cancel_text)?.try_into_option()?;
+
+        let main_params = ConfirmBlobParams::new(
```

Comment:
`ConfirmBlobParams::new` is used with lots of `None`s and `false` in this function. I consider removing these from `new` and instantiate them only using `with_xyz` constructs.